lib/Driver/Job.cpp

```diff
@@ -352,7 +352,11 @@ void Job::printCommandLineAndEnvironment(raw_ostream &Stream,
 void Job::printCommandLine(raw_ostream &os, StringRef Terminator) const {
   escapeAndPrintString(os, Executable);
   os << ' ';
-  printArguments(os, Arguments);
+  if (hasResponseFile()) {
+    printArguments(os, ResponseFileArg);
```

Comment:
This seems wrong; we'd want to dump the regular command line, but note that the job will be using a response file (similar to how we do environment variables).

Reply:
I'm not exactly sure how the output should be formatted (I don't see environment variables printed when I use `-v` or `-debug-crash-immediately`). As written, it would look similar to:
```
/usr/bin/swift @/tmp/arguments-70c499.resp
```
A possible alternative would be:
```
[Arguments: @/tmp/arguments-70c499.resp]
/usr/bin/swift -frontend -c -primary-file main.swift test.swift -target x86_64-unknown-linux-gnu -disable-objc-interop -color-diagnostics -module-name test -o /tmp/main-6d68e0.o
```
Also, wouldn't you run into the same issue when filelists are used?

Reply:
That's true, but filelists hold *much* less information. (Though it's still important enough that the frontend jumps through hoops to show them in the PrettyStackTrace.)

Reply:
To clarify, is my suggested alternative sufficient?

Reply:
Ah, sorry. The idea is, but we still want one command per line. I could see either

```
/usr/bin/swift @/tmp/arguments-70c499.resp # -frontend -c -primary-file main.swift …
```

or

```
/usr/bin/swift -frontend -c -primary-file main.swift … # [Arguments: @/tmp/arguments-70c499.resp]
```

What do you think?

Reply:
I prefer the first of your two options since it is literally what is being executed. Plus, a user could actually execute the first one, whereas the second would likely fail due to too many arguments.

Reply:
I am looking at how to implement this for the stack trace during a crash, and ran into some issues. Printing out the program and its arguments during a crash is handled by `llvm::PrettyStackTrace`, and it just loops over the `argv`. The problem is that the `argv` is initialized in the first line of `main` in `tools/driver/driver.cpp`. At this point, the executable has no knowledge of the command line arguments if a response file is being used. I would either 1) move `PROGRAM_START` after the response files are expanded, or 2) modify the internals of `PrettyStackTrace` to expand any response files. It looks like `clang` initializes the stack trace before expanding response files: https://github.com/apple/swift-clang/blob/stable/tools/driver/driver.cpp#L326

Reply:
I think I'd prefer (1). AFAIK Clang never uses response files to communicate between the driver and frontend, and those are the most important crash logs. @davidungar?

Reply:
I should point out that if I move `PROGRAM_START`, the response file name and path will not be printed, since it is consumed when expanded.